mixins/loading-complete/loading-complete-mixin.js

```diff
@@ -1,21 +1,36 @@
-export const LoadingCompleteMixin = (superclass) => class extends superclass {
+import { dedupeMixin } from '@open-wc/dedupe-mixin';
 
-	constructor() {
-		super();
-		this._loadingCompletePromise = new Promise(resolve => {
-			this._loadingCompleteResolve = resolve;
-		});
+export const LoadingCompleteMixin = dedupeMixin((superclass) => class extends superclass {
+
+	#loadingCompleteResolve;
+
+	// eslint-disable-next-line sort-class-members/sort-class-members
+	#loadingCompletePromise = this.getLoadingComplete === this.#getLoadingComplete
+		? new Promise(resolve => this.#loadingCompleteResolve = resolve)
+		: Promise.resolve();
+
+	get getLoadingComplete() {
+		return this.#getLoadingComplete;
+	}
+
+	get loadingComplete() {
+		return this.#getLoadingComplete();
+	}
+
+	get resolveLoadingComplete() {
+		return this.#resolveLoadingComplete.bind(this);
```

Comment:
This makes it unnecessary to wrap before calling.

```js
this.getData.then(this.resolveLoadingComplete);
```

Reply:
Why not use an arrow capture here?
```js
return ()=> this.#resolveLoadingComplete();
```
Not sure if there's a draw back/better reason to using bind or `() =>`

Reply:
In this case, I don't think there's a difference. Just an old habit to use `bind` in this scenario.

Reply:
what's the difference (functional/semantic/otherwise) between this vs. just keeping `#resolveLoadingComplete` public?

Reply:
ok right yea that makes sense, fresh eyes help me read better